test/cag-common-transforms.js

```diff
@@ -68,11 +69,50 @@ test('CAG should intersect properly', t => {
   const lastSide = result.sides[result.sides.length - 1]
 
   t.deepEqual(result.sides.length, 5)
-  t.deepEqual(compactCagSide(firstSide), {pos: [[0.30901699437494723, -0.9510565162951536], [1, 0]]})
-  t.deepEqual(compactCagSide(lastSide), {pos: [[-0.8090169943749475, -0.587785252292473], [0.30901699437494723, -0.9510565162951536]]})
+  t.deepEqual(compareCagSide(firstSide), {pos: [[0.30901699437494723, -0.9510565162951536], [1, 0]]})
+  t.deepEqual(compareCagSide(lastSide), {pos: [[-0.8090169943749475, -0.587785252292473], [0.30901699437494723, -0.9510565162951536]]})
 })
 
-test.todo('CAG should transform properly')
-test.todo('CAG should flip properly')
-test.todo('CAG should expand properly')
-test.todo('CAG should contract properly')
+test('CAG should transform properly', t => {
+  const cag1 = CAG.rectangle({radius: 3})
+  const matrixA = CSG.Matrix4x4.translation([10, 10, 0])
+
+  let res1 = cag1.transform(matrixA)
+  t.deepEqual(compareCagSide(res1.sides[0]), {pos: [[7, 13], [7, 7]]})
+  t.deepEqual(compareCagSide(res1.sides[1]), {pos: [[7, 7], [13, 7]]})
+  t.deepEqual(compareCagSide(res1.sides[2]), {pos: [[13, 7], [13, 13]]})
+  t.deepEqual(compareCagSide(res1.sides[3]), {pos: [[13, 13], [7, 13]]})
+})
+
+test('CAG should flip properly', t => {
+  const cag1 = CAG.rectangle({center: [5,-5],radius: [3,2]})
+  let res1 = cag1.flipped()
+  t.deepEqual(compareCagSide(res1.sides[0]), {pos: [[2, -3], [8, -3]]})
+  t.deepEqual(compareCagSide(res1.sides[1]), {pos: [[8, -3], [8, -7]]})
+  t.deepEqual(compareCagSide(res1.sides[2]), {pos: [[8, -7], [2, -7]]})
+  t.deepEqual(compareCagSide(res1.sides[3]), {pos: [[2, -7], [2, -3]]})
+})
+
+test.failing('CAG should expand properly', t => {
+  const cag1 = CAG.rectangle({center: [5,-5],radius: [5,3]})
+//console.log(cag1.toString())
+  let res1 = cag1.expand(1, 4)
+//console.log(res1.toString())
+// FIX ME order of sides is just WRONG!!!!
+  t.deepEqual(compareCagSide(res1.sides[0]), {pos: [[11, -8], [11, -2]]})
```

Comment:
Why is it wrong ? Does it prevent us from testing the expansion feature ?

Reply:
It's not even close... after this, the sides are following some strange order. The sides should be counterclockwise otherwise later changes will produce wrong results, such as extrude.

This may be a fundamental issue with CAG.

Reply:
ouch that is really bad you are right ! this also impacts any feature using this one under the hood ! Created issue #66 to cover this